core/common/src/ContourDisplay.ts

```diff
@@ -0,0 +1,368 @@
+  public readonly pattern: LinePixels;
+
+  public equals(other: ContourStyle): boolean {
+    if (!this.color.equals(other.color) || this.pixelWidth !== other.pixelWidth || this.pattern !== other.pattern) {
+      return false;
+    }
+    return true;
+  }
+
+  /** Compares two contour styles.
+   * @param lhs First contour style to compare
+   * @param rhs Second contour style to compare
+   * @returns 0 if lhs is equivalent to rhs, a negative number if lhs compares less than rhs, or a positive number if lhs compares greater than rhs.
+   * @public
+   */
+  public static compare(lhs: ContourStyle, rhs: ContourStyle): number {
+    let diff = 0;
+    if ((diff = lhs.color.compareTo(rhs.color)) !== 0)
+      return diff;
+    if ((diff = compareNumbers(lhs.pixelWidth, rhs.pixelWidth)) !== 0)
+      return diff;
+    if ((diff = compareNumbers(lhs.pattern, rhs.pattern)) !== 0)
+      return diff;
+
+    return diff;
+  }
+
+  private constructor(json?: ContourStyleProps) {
+    if (undefined === json) {
+      this.color = RgbColor.fromJSON({r: 0, g: 0, b: 0});
+      this.pixelWidth = 1;
+      this.pattern = 0;
+    } else {
+      this.color = json.color ? RgbColor.fromJSON(json.color) : RgbColor.fromJSON({r: 0, g: 0, b: 0});
+      this.pixelWidth = json.pixelWidth ?? 1;
+      this.pattern = json.pattern ?? 0;
+    }
+  }
+
+  public static fromJSON(json?: ContourStyleProps) {
+    return json ? new ContourStyle(json) : new ContourStyle({});
+  }
+
+  public toJSON(): ContourStyleProps {
+    const props: ContourStyleProps = {};
+
+    if (!this.color.equals(RgbColor.fromJSON({r: 0, g: 0, b: 0})))
+      props.color = this.color.toJSON();
+
+    if (1 !== this.pixelWidth)
+      props.pixelWidth = this.pixelWidth;
+
+    if (0 !== this.pattern)
+      props.pattern = this.pattern;
+
+    return props;
+  }
+}
+
+/** JSON representation of the rendering styling settings that apply to a specific set of subcategories within a [[ContourGroup]].
+ * @public
+ */
+export interface ContourProps {
+  /** See [[Contour.majorStyle]]. */
+  majorStyle?: ContourStyleProps;
+  /** See [[Contour.minor]]. */
+  minorStyle?: ContourStyleProps;
+  /** See [[Contour.minorInterval]]. */
+  minorInterval?: number;
+  /** See [[Contour.majorIntervalCount]]. */
+  majorIntervalCount?: number;
+  /** See [[Contour.showGeometry]] */
+  showGeometry?: boolean;
+}
+
+/** The rendering styling settings that apply to a specific set of subcategories within a [[ContourGroup]].
+ * This actually describes stylings for two sets of contours: major and minor. These stylings are separate from each other.
+ * The minor contour occurs at a defined interval in meters. These intervals draw at a fixed height; they are not dependent on the range of the geometry to which they are applied.
+ * The major contour is dependent on the minor contour. The interval of its occurence is not measured directly in meters; rather it is a count of minor contour intervals between its occurrences.
```

Comment:
> it is a count of minor contour intervals between its occurrences.

Is this accurate? i.e.,:
majorInterval = 0 means every contour is major,
1 means every other contour is major, and
2 means every third contour is major?
I thought I recalled from your demos that every nth contour was major, where n = majorInterval.

Reply:
Thanks for the catch - you are right; this is unclear and indicating the wrong thing.  Your understanding from the demo is the correct one.  I pushed a different explanation to both the .md and the .ts.